FXyz-Core/src/main/java/org/fxyz3d/shapes/polygon/PolygonMesh.java

```diff
@@ -41,17 +41,20 @@
  * can convert to using ObservableIntegerArray
  */
 public class PolygonMesh {
-    private final ObservableFloatArray points = FXCollections.observableFloatArray();
-    private final ObservableFloatArray texCoords = FXCollections.observableFloatArray();
+    private final ObservableFloatArray points;
+    private final ObservableFloatArray texCoords;
     public int[][] faces = new int[0][0];
     private final ObservableIntegerArray faceSmoothingGroups = FXCollections.observableIntegerArray();
     protected int numEdgesInFaces = -1; // TODO invalidate automatically by listening to faces (whenever it is an observable)
 
-    public PolygonMesh() {}
+    public PolygonMesh() {
```

Comment:
Maybe use these constructors?

```
private final ObservableFloatArray points;
    private final ObservableFloatArray texCoords;
    public int[][] faces;
   ... 
    public PolygonMesh() {
        this(FXCollections.observableFloatArray(), FXCollections.observableFloatArray(), new int[0][0]);
    }

    public PolygonMesh(float[] points, float[] texCoords, int[][] faces) {
        this(FXCollections.observableFloatArray(points), FXCollections.observableFloatArray(texCoords), faces);
    }

    public PolygonMesh(ObservableFloatArray points, ObservableFloatArray texCoords, int[][] faces) {
        this.points = points;
        this.texCoords = texCoords;
        this.faces = faces;
    }
```

Reply:
I tried not to touch classes outside of `ObjImporter`. e.g., I don't see a reason for `int[][] faces` to be `public`. As for the constructors, it looks like only the empty constructor is being called anyway. Do you think these other constructors will be used?

Reply:
It's better to have more options, imo, than to restrict them because we don't have a use case for it yet.

About faces being public, I don't like that either. Probably it can be private (or protected) with a public setter/getter?